src/service-worker/core.js

```diff
@@ -259,6 +270,36 @@ self.addEventListener('fetch', event => {
       return fetchAndCache(cache, versionedRequest, requestFile, version);
     });
   });
+}
+
+// Setup the Fetch listener, for when the client is on the CDN origin.
+self.addEventListener('fetch', event => {
+  const response = handleFetch(event.request, event.clientId);
+
+  // We only get a response promise back if it's a request we care to cache.
+  if (!response) {
+    return;
+  }
 
   event.respondWith(response);
 });
+
+// Setup the Foreign Fetch listener, for when the client is on a Publisher
+// origin.
+self.addEventListener('foreignfetch', event => {
+  const response = handleFetch(event.request, event.clientId);
+
+  // We only get a response promise back if it's a request we care to cache.
+  if (!response) {
+    return;
+  }
+
+  event.respondWith(response.then(response => {
```

Comment:
Nit: I think this would read clearer if you used a variable name other than `response` for either the initial promise or the parameter inside the `.then()`.